### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by dsteinich**

### CODE REVIEW

File: src/test/java/gov/usgs/wma/mlrlegacy/db/ControllerRIT.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import org.junit.Test;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+import org.springframework.http.ResponseEntity;
+
+import com.github.springtestdbunit.annotation.DatabaseSetup;
+
+@DatabaseSetup("classpath:/testData/setupOne/")
+public class ControllerRIT extends BaseControllerIT {
```
Comment (mbucknell) [inline src/test/java/gov/usgs/wma/mlrlegacy/db/ControllerRIT.java]: Curious as to why you created different classes for  the 'R' Tests and the 'CU' tests?

File: src/test/resources/testData/setupOne/table-ordering.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+legacy_location
```
Comment (mbucknell) [inline src/test/resources/testData/setupOne/table-ordering.txt]: I couldn't find a good explanation of why this is needed. Can you clarify what this is used for.

File: src/test/java/gov/usgs/wma/mlrlegacy/db/ControllerRIT.java (hunk 1 above)
Comment (dsteinich) [reply src/test/java/gov/usgs/wma/mlrlegacy/db/ControllerRIT.java]: Purely performance related - The 'R' (read-only) tests can use the same data for all the tests and do not modify it, so the database only gets loaded once for all tests in that class. The 'CU' tests modify the data, so we must reload the database before each test.

File: src/test/resources/testData/setupOne/table-ordering.txt (hunk 2 above)
Comment (dsteinich) [reply src/test/resources/testData/setupOne/table-ordering.txt]: Interesting that it really isn't well documented, but fails quickly if you don't have it...
When loading from csv files, it instructs DBUnit which order to process the tables in and which *.csv files to use. The csv convention requires one file per database table.
If using xml, you can put all of the data into one file, and order it in the file appropriately...

**Pull request merged**